lib/workers/repository/dependency-dashboard.ts

```diff
@@ -224,8 +224,29 @@ export async function ensureDependencyDashboard(
     return;
   }
   logger.debug('Ensuring Dependency Dashboard');
+
+  // Check packageFiles for any deprecations
+  let hasDeprecations = false;
+  const deprecatedPackages: Record<string, Record<string, boolean>> = {};
+  logger.debug({ packageFiles }, 'Checking packageFiles for deprecations');
+  for (const [manager, fileNames] of Object.entries(packageFiles)) {
+    for (const fileName of fileNames) {
+      for (const dep of fileName.deps) {
+        const name = dep.packageName ?? dep.depName;
+        const hasReplacement = !!dep.updates?.find(
+          (updates) => updates.updateType === 'replacement',
+        );
+        if (name && (dep.deprecationMessage ?? hasReplacement)) {
+          hasDeprecations = true;
+          deprecatedPackages[manager] ??= {};
+          deprecatedPackages[manager][name] ??= hasReplacement;
+        }
+      }
+    }
+  }
```

Comment:
Wouldn't it make sense to move this out of dependency dashboard logic so that this information is available in other parts? E.g. for later packageRules.

Reply:
Ultimately this logic is building up a list of deprecations for the entire repo, which then isn't useful for packageRules. The only useful part might be to do this:
- Add a deprecationMessage for the cases where a replacement update is available but the package isn't flagged as deprecated
- Add a new option to packageRules for matching isDeprecated

But I don't want this important feature to wait on either of those, I think they can come later. This code is pretty modular and can be refactored easily